mdata/reorder_buffer_test.go

```diff
@@ -0,0 +1,333 @@
+package mdata
+
+import (
+	"fmt"
+	"reflect"
+	"testing"
+
+	"gopkg.in/raintank/schema.v1"
+)
+
+func testAddAndGet(t *testing.T, reorderWindow uint32, testData, expectedData []schema.Point, expectAddFail bool) []schema.Point {
+	var flushed []schema.Point
+	b := NewReorderBuffer(reorderWindow, 1)
+	metricsTooOld.SetUint32(0)
+	for _, point := range testData {
+		addRes := b.Add(point.Ts, point.Val)
+		flushed = append(flushed, addRes...)
+	}
+	if expectAddFail && metricsTooOld.Peek() == 0 {
```

Comment:
why don't we just make expectAddFail an int, that way you can just check `expectedFails != metricsTooOld.Peek()` which is simpler + allows for tests that want multiple failures